test/box-luatest/session_test.lua

```diff
@@ -0,0 +1,311 @@
+        t.assert(not ok, 'check access')
+        conn:close()
+        conn = nil
+        while not on_disconnect_user do fiber.sleep(0.001) end
+        -- Triggers are executed with admin permissions.
+        t.assert_equals(on_connect_user, 'admin',
+                        'check trigger permissions, on_connect')
+        t.assert_equals(on_disconnect_user, 'admin',
+                        'check trigger permissions, on_disconnect')
+
+        box.session.on_connect(nil, on_connect)
+        box.session.on_disconnect(nil, on_disconnect)
+    end)
+end
+
+-- Check session privilege.
+g.test_session_priveleges = function(cg)
+    cg.server:exec(function()
+        local net_box = require('net.box')
+        local uri = box.cfg.listen
+        local ok = pcall(function()
+            net_box.connect(uri)
+        end)
+        t.assert(ok, 'session privilege')
+        box.schema.user.create('tester', { password = 'tester' })
+        box.schema.user.revoke('tester', 'session', 'universe')
+        local conn = net_box.connect('tester:tester@' .. uri)
+        t.assert(conn.state, 'error', 'session privilege state')
+        t.assert(conn.error:match('Session'), 'sesssion privilege errmsg')
+        ok = pcall(box.session.su, 'user1')
+        t.assert(not ok, 'session.su on revoked')
+        box.schema.user.drop('tester')
+    end)
+end
+
+g.test_session_gh_2994 = function(cg)
+    local uri = cg.server.net_box_uri
+    local net_box = require('net.box')
+    local conn = net_box.connect(uri)
+    t.assert(conn:eval('return box.session.exists(box.session.id())'),
+             'remote session exist check')
+    t.assert_not_equals(
+        conn:eval('return box.session.peer(box.session.id())'),
+        nil, 'remote session peer check')
+    t.assert(conn:eval([[
+return box.session.peer() == box.session.peer(box.session.id())]]),
+             'remote session peer check')
+    t.assert(conn:eval([[
+return box.session.fd() == box.session.fd(box.session.id())]]),
+             'remote session fd check')
+end
+
+-- gh-2994: Session UID vs session EUID.
+g.test_session_gh_2994 = function(cg)
+    cg.server:exec(function()
+    local uri = box.cfg.listen
+    local net_box = require('net.box')
+    local conn = net_box.connect(uri)
+    -- t.assert_equals(box.session.euid(), 1, 'session.uid')
+    -- t.assert_equals(box.session.su('guest', box.session.uid), 1,
+    --                 'session.uid from su is admin')
+    -- t.assert_equals(box.session.su('guest', box.session.euid), 0,
+    --                 'session.euid from su is guest')
+    local id = conn:eval('return box.session.uid()')
+    t.assert_equals(id, 0, 'session.uid from netbox')
+    id = conn:eval('return box.session.euid()')
+    t.assert_equals(id, 0, 'session.euid from netbox')
+    box.session.su('admin')
+    conn:eval('box.session.su(\"admin\", box.schema.create_space, \"sp1\")')
+    local sp = conn:eval('return box.space._space.index.name:get{\"sp1\"}[2]')
```

Comment:
Minor: We don't need escaping double quotes in these 2 lines.

Reply:
Fixed.